File: opcua-webservice-spring/tutorials/lesson3/src/main/java/org/opcfoundation/uawebservicedemo/models/employee/employeedata/event/EmployeeCheckInEventType.java
package org.opcfoundation.uawebservicedemo.models.employee.employeedata.event;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.uawebservicedemo.database.entity.EmployeeCheckIn;
import org.opcfoundation.webserver.addressspace.nodes.builtin.UaDataTypes;
import org.opcfoundation.webserver.digitaltwin.DigitalTwinSpace;
import org.opcfoundation.webserver.digitaltwin.event.EventData;
import org.opcfoundation.webserver.digitaltwin.event.EventType;

import java.time.LocalDateTime;

public class EmployeeCheckInEventType extends EventType {
    public static final String CheckIn = "CheckIn";
    public static final String Location = "Location";

    public EmployeeCheckInEventType(DigitalTwinSpace space)
    {
        super("EmployeeCheckInEventType", new LocalizedText("EmployeeCheckInEventType"), space);
        addField(CheckIn, new LocalizedText(CheckIn), new LocalizedText("Is check in"), UaDataTypes.Boolean);
        addField(Location, new LocalizedText(Location), new LocalizedText("Location"), UaDataTypes.String);
    }

    public static EventData generateEventData(EmployeeCheckIn record)
    {
        EventData data = new EventData(record.getId().toString());
        data.setTime(record.getTime());
        data.setMessage(record.getRemark());
        data.setFieldData(CheckIn, Variant.ofBoolean(record.getCheckIn()));
        data.setFieldData(Location, Variant.ofString(record.getLocation()));

        return data;
    }
}

File: opcua-webservice-node/examples/demoserver/src/models/employee/employeedata/EmployeeCheckInEventType.ts
import { UaLocalizedText, UaVariant } from "opcua-webapi-ts";
import { EventData, EventType, UaDataTypes } from "opcua-webservice-node";

import type { employeecheckin } from "../../../connectors/prisma/client";
import { EmployeeTwinSpace } from "../../EmployeeTwinSpace";

export class EmployeeCheckInEventType extends EventType {
    static readonly CheckIn = "CheckIn";
    static readonly Location = "Location";

    constructor(space: EmployeeTwinSpace) {
        super("EmployeeCheckInEventType", new UaLocalizedText("EmployeeCheckInEventType"), space);

        this.addField(
            EmployeeCheckInEventType.CheckIn,
            new UaLocalizedText(EmployeeCheckInEventType.CheckIn),
            new UaLocalizedText("Is check in"),
            UaDataTypes.Boolean,
        );

        this.addField(
            EmployeeCheckInEventType.Location,
            new UaLocalizedText(EmployeeCheckInEventType.Location),
            new UaLocalizedText("Location"),
            UaDataTypes.String,
        );
    }

    static generateEventData(record: employeecheckin): EventData {
        const data = new EventData(record.ID.toString());
        data.time = record.Time;
        data.message = record.Remark ?? "";
        data.setFieldData(EmployeeCheckInEventType.CheckIn, UaVariant.boolean(record.CheckIn));
        data.setFieldData(EmployeeCheckInEventType.Location, UaVariant.string(record.Location));

        return data;
    }
}